window/src/os/x11/connection.rs

```diff
@@ -239,12 +239,22 @@ impl ConnectionOps for XConnection {
             .ok_or_else(|| anyhow::anyhow!("no screens were found"))?
             .clone();
 
-        // We don't yet know how to determine the active screen,
-        // so assume the main screen.
-        // TODO: find focused window and resolve it!
-        // Maybe something like <https://stackoverflow.com/a/43666928/149111>
-        // but ported to Rust?
-        let active = main.clone();
+        self.conn.send_request(&xcb::x::GrabServer {});
+        let focused = self
+            .send_and_wait_request(&xcb::x::GetInputFocus {})
+            .context("querying focused window")?;
+        let geom = self
+            .send_and_wait_request(&xcb::x::GetGeometry {
+                drawable: xcb::x::Drawable::Window(focused.focus()),
+            })
+            .context("querying geometry")?;
+        let window_origin = ScreenPoint::new(geom.x().into(), geom.y().into());
+        let active = by_name
+            .values()
+            .find(|screen| screen.rect.contains(window_origin))
+            .ok_or_else(|| anyhow::anyhow!("active window is not in any screen"))?
+            .clone();
```

Comment:
I think it's technically possible that no windows have focus, so perhaps this bit should look something like:

```rust
let active = self.screen_from_focused_window(&by_name).unwrap_or_else(|| main.clone());
```

with the heavy lifting moved into a helper method:

```rust
fn screen_from_focused_window(&self, by_name: &HashMap<String,ScreenInfo>) -> anyhow::Result<ScreenInfo> {
       let focused = self
            .send_and_wait_request(&xcb::x::GetInputFocus {})
            .context("querying focused window")?;
        let geom = self
            .send_and_wait_request(&xcb::x::GetGeometry {
                drawable: xcb::x::Drawable::Window(focused.focus()),
            })
            .context("querying geometry")?;
        let window_origin = ScreenPoint::new(geom.x().into(), geom.y().into());
        Ok(by_name
            .values()
            .find(|screen| screen.rect.contains(window_origin))
            .ok_or_else(|| anyhow::anyhow!("active window is not in any screen"))?
            .clone())
}
```